src/pandas_profiling/model/typeset.py

```diff
@@ -191,10 +192,50 @@ def get_relations(cls):
 
     @classmethod
     @nullable_series_contains
-    def contains_op(cls, series: pd.Series, state: dict) -> bool:
+    def contains_op(cls, series: pd.Series) -> bool:
         return pdt.is_complex_dtype(series)
 
 
+class SparkUnsupported(visions.VisionsBaseType):
+    pass
+
+
+class SparkNumeric(visions.VisionsBaseType):
+    @classmethod
+    def get_relations(cls):
+        return [IdentityRelation(cls, SparkUnsupported)]
+
+    @classmethod
+    def contains_op(cls, series: SparkSeries, state: dict) -> bool:
+        return (
+            str(series.type) == "DoubleType"

```

Comment:
`str(series.type) in ["DoubleType", ...]` would be cleaner and faster

Reply:
changed